interfaces/core.go

```diff
@@ -77,6 +77,9 @@ func (conn *ConnRef) ID() string {
 	return fmt.Sprintf("%s:%s %s:%s", conn.PlugRef.Snap, conn.PlugRef.Name, conn.SlotRef.Snap, conn.SlotRef.Name)
 }
 
+// Snippet is an opaque type for conveying typed information between interfaces and backends.
+type Snippet interface{}
```

Comment:
This may be dropped for the time being. We don't really want to make that hack wider, but rather fix the implementation as we discussed.